CRM/Financial/Form/BatchTransaction.php

```diff
@@ -52,7 +58,14 @@ public function preProcess() {
     $this->assign('entityID', self::$_entityID);
     if (isset(self::$_entityID)) {
       $this->_batchStatusId = CRM_Core_DAO::getFieldValue('CRM_Batch_BAO_Batch', self::$_entityID, 'status_id');
+      $batchStatuses = CRM_Core_PseudoConstant::get('CRM_Batch_DAO_Batch', 'status_id', array('labelColumn' => 'name', 'condition' => " v.value={$this->_batchStatusId}"));
+      $this->_batchStatus = $batchStatuses[$this->_batchStatusId];
       $this->assign('statusID', $this->_batchStatusId);
+      $this->assign('batchStatus', $this->_batchStatus);
+      $this->assign('validStatus', FALSE);
+      if (in_array($this->_batchStatus, array('Open', 'Reopened'))) {
+        $this->assign('validStatus', TRUE);
+      }
```

Comment:
Might be better to set to a variable and assign the variable to template rather than assigning twice if the *if* condition passes.